doc/source/data/execution-configurations.rst

```diff
@@ -0,0 +1,54 @@
+.. _execution_configurations:
+
+=================================
+Execution Configurations
+=================================
+
+Ray Data provides a number of configurations that can be used to control various aspects
+of Ray Dataset execution. These configurations can be modified by the user using
+:class:`~ray.data.ExecutionOptions` and :class:`~ray.data.DataContext`. 
+This guide describes the most important of these configurations and when to use them.
+
+Configuring :class:`~ray.data.ExecutionOptions`
+===============================================
+
+The :class:`~ray.data.ExecutionOptions` class is used to configure options during Ray Dataset execution.
+To use it, you can modify the attributes in the current :class:`~ray.data.DataContext` object's `execution_options`. For example:
+
+.. code-block::
+
+   ctx = ray.data.DataContext.get_current()
+   ctx.execution_options.verbose_progress = True
+
+* `resource_limits`: Set a soft limit on the resource usage during execution. Auto-detected by default.
```

Comment:
When would you want to set a limit on resource usage?

Reply:
added an example of such a case:
> For example, if there are other parts of the code which require some minimum amount of resources, you may want to limit the amount of resources that Ray Data uses.

Reply:
For my own understanding, when would you want to use `resorce_limits` over `exclude_resources` if you have other code that uses resources?

Reply:
i think they are two ways of controlling the same overall concept. one is an exclusion of resources used by non-ray data workload, while the other is a cap on data resources.